### PR TITLE
fix(macOS Button): Use Fluent 2 alias tokens

### PR DESCRIPTION
**Pull request by Saadnajmi**
### Platforms Impacted
- [ ] iOS
- [x] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

The macOS Button color tokens were still being set by older, deprecated tokens instead of the Fluent-2-vetted alias tokens. Let's update them.

#### Notable changes and things I learned (though I probably missed a few):

- macOS default buttons use neutralBackground**2** / neutralForeground**2** /  neutralstroke**2** for a softer overall color set
- macOS in general shouldn't have a hover state for its' buttons, except Subtle Button. Some alias tokens don't respect this, see the followup tasks.
- Apart from Subtle Button, I try to always define a non-transparent border, which adds 2 px to the overall width and height of each button.  Subtle Button has bugs, see the followup tasks.
- strokeFocus1 and strokeFocus2 seem very win32 specific, so we probably shouldn't use them elsewhere. They seem explicitly for the two-toned focus border. 

#### Followups:

- A PR to delete the old `default<Token> / ghost<Token> / etc ramp
- https://github.com/microsoft/fluentui-react-native/issues/3780
- https://github.com/microsoft/fluentui-react-native/issues/3781
- https://github.com/microsoft/fluentui-react-native/issues/3782

### Verification

Video of how the buttons look now.

https://github.com/user-attachments/assets/e1ab7fc0-8980-4300-81c0-b62cda4d0664

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

**Comment by rurikoaraki** [comment]
"and that our jest tests are actually testing that on macOS."

-- I think right now our jest tests only run on one platform at a time, so we kinda have to pick which platform we want tested.

File: packages/components/Button/src/ButtonColorTokens.macos.ts [hunk 1]
```diff
@@ -3,94 +3,101 @@ import type { TokenSettings } from '@fluentui-react-native/use-styling';
 
 import type { ButtonTokens } from './Button.types';
 
+// Github #X: We should be using the `hover` tokens for the hover states, but they currently set a
+// different color from rest. macOS by default doesn't have many hover states on it's controls. Address
```
Comment (rurikoaraki) [inline packages/components/Button/src/ButtonColorTokens.macos.ts:7]: grammar nit: its not it's

File: packages/components/Button/src/ButtonColorTokens.macos.ts [hunk 2]
```diff
@@ -3,94 +3,101 @@ import type { TokenSettings } from '@fluentui-react-native/use-styling';
...
+// this discrepancy in the token packages, and then update these definitions as appropriate.
+
+// Github #X2: The macOS brand ramps from the token package use blue from Fluent V1, instead of communication blue.
```
[the remark this answers is not in the document]
Comment (Saadnajmi) [reply packages/components/Button/src/ButtonColorTokens.macos.ts:10]: Indeed. I think to align with Fluent 2 (and web and mobile as a consequence), I’d want to update to use commblue. My understanding is there was a lot more color science work done in creating the fluent 2 band ramp.

**Review by rurikoaraki — APPROVED** [review]

**Comment by Saadnajmi** [comment]
Moving back to draft, as I am making updates to the token definitions

**Comment by Saadnajmi** [comment]
Token definition updates blocked upstream. Merging as is.

**Pull request merged**